src/SyntaxHighlighter/X86Assembly.cpp

```diff
@@ -216,9 +229,34 @@ const QRegularExpression kPlatformRegex{"^Platform:.*$"};
 X86Assembly::X86Assembly() : QSyntaxHighlighter{static_cast<QObject*>(nullptr)} {}
 
 void X86Assembly::highlightBlock(const QString& code) {
-  HighlightBlockAssembly(code, [this](int start, int count, const QTextCharFormat& format) {
-    setFormat(start, count, format);
-  });
+  const auto* const metadata = static_cast<const Metadata*>(currentBlock().userData());
```

Comment:
If we go with currentBlock().userData() you also have to check whether it returns a null pointer.

If only main content is set, `userData` is not used. It will only be populated when there are also annotating lines.